docs/smart-contracts/meebits_erc721.sol

```diff
@@ -0,0 +1,679 @@
+    string public contentHash = "QmfXYgfX1qNfzQ6NRyFnupniZusasFPMeiWn5aaDnx7YXo";
+
+    uint public constant TOKEN_LIMIT = 20000;
+    uint public constant SALE_LIMIT = 9000;
+
+    mapping(bytes4 => bool) internal supportedInterfaces;
+
+    mapping (uint256 => address) internal idToOwner;
+
+    mapping (uint256 => uint256) public creatorNftMints;
+
+    mapping (uint256 => address) internal idToApproval;
+
+    mapping (address => mapping (address => bool)) internal ownerToOperators;
+
+    mapping(address => uint256[]) internal ownerToIds;
+
+    mapping(uint256 => uint256) internal idToOwnerIndex;
+
+    string internal nftName = "Meebits";
+    string internal nftSymbol = unicode"⚇";
+
+    uint internal numTokens = 0;
+    uint internal numSales = 0;
+
+    // Cryptopunks contract
+    address internal punks;
+
+    // Autoglyphs contract
+    address internal glyphs;
+
+    address payable internal deployer;
+    address payable internal beneficiary;
+    bool public communityGrant = true;
+    bool public publicSale = false;
+    uint private price;
+    uint public saleStartTime;
+    uint public saleDuration;
+
+    //// Random index assignment
+    uint internal nonce = 0;
+    uint[TOKEN_LIMIT] internal indices;
+
+    //// Market
+    bool public marketPaused;
+    bool public contractSealed;
+    mapping (address => uint256) public ethBalance;
+    mapping (bytes32 => bool) public cancelledOffers;
+
+    modifier onlyDeployer() {
+        require(msg.sender == deployer, "Only deployer.");
+        _;
+    }
+
+    bool private reentrancyLock = false;
+
+    /* Prevent a contract function from being reentrant-called. */
+    modifier reentrancyGuard {
+        if (reentrancyLock) {
+            revert();
+        }
+        reentrancyLock = true;
+        _;
+        reentrancyLock = false;
+    }
+
+    modifier canOperate(uint256 _tokenId) {
+        address tokenOwner = idToOwner[_tokenId];
+        require(tokenOwner == msg.sender || ownerToOperators[tokenOwner][msg.sender], "Cannot operate.");
+        _;
+    }
+
+    modifier canTransfer(uint256 _tokenId) {
+        address tokenOwner = idToOwner[_tokenId];
+        require(
+            tokenOwner == msg.sender
+            || idToApproval[_tokenId] == msg.sender
+            || ownerToOperators[tokenOwner][msg.sender], "Cannot transfer."
+        );
+        _;
+    }
+
+    modifier validNFToken(uint256 _tokenId) {
+        require(idToOwner[_tokenId] != address(0), "Invalid token.");
+        _;
+    }
+
+    constructor(address _punks, address _glyphs, address payable _beneficiary) {
+        supportedInterfaces[0x01ffc9a7] = true; // ERC165
+        supportedInterfaces[0x80ac58cd] = true; // ERC721
+        supportedInterfaces[0x780e9d63] = true; // ERC721 Enumerable
+        supportedInterfaces[0x5b5e139f] = true; // ERC721 Metadata
+        deployer = msg.sender;
+        punks = _punks;
+        glyphs = _glyphs;
+        beneficiary = _beneficiary;
+    }
```

Comment:
deployer is account creating the contract and is given additional rights for example to pause the market. 
beneficiary is the adress of the developer larva labs and used for the proceeds of the minting (called from mint() ).
punks and glyphs points to the addresses for those contracts .
supportedInterfaces used for supportsInterface (ref ERC165 - Standard Interface Detection)

Reply:
The constructor arguments can be found at https://etherscan.io/address/0x7bd29408f11d2bfc23c34f18275bbf23bb716bc7#code
-----Decoded View---------------
Arg [0] : _punks (address): 0xb47e3cd837ddf8e4c57f05d70ab865de6e193bbb
Arg [1] : _glyphs (address): 0xd4e4078ca3495de5b1d4db434bebc5a986197782
Arg [2] : _beneficiary (address): 0xc352b534e8b987e036a93539fd6897f53488e56a

Reply:
Deployer for meebits is 0xc352b534e8b987e036a93539fd6897f53488e56a 
Found from the first transaction of the contract
https://etherscan.io/tx/0xf2040b9b67193fe8c861a18cff864b9f35c1f69cc8734c724c388c449a1116c4 

![image](https://user-images.githubusercontent.com/1133607/120157685-91aa4180-c1f3-11eb-84fa-11a4138c9abf.png)